_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+    -   What losses, challenges, and obstacles did we encounter?
+    -   What did we learn?
+-   Frequency: quarterly
+-   Responsible: VP Engineering
+-   Contributors: project leads and other team members
+-   Products:
+    -   Ranked list of things that worked well and things that didn't
+    -   Ranked list of ideas for things to try to adopt, improve, or drop
+
+### Engineering Quarterly Retrospective
+
+Similarly,
+we have a general retrospective at the end of each quarter
+to ask ourselves wider-ranging questions
+that may not be specific to a particular project's goals.
+Practices like having an on-call user support rotation
+have grown out of these self-reviews.
+
+-   Summary: End of quarter team sync to reflect on the quarter and identify ways to improve how we work and our ability to deliver
+-   Key Questions:
+    -   What went well?
+    -   What didn't?
+    -   What changes would we like to implement to improve the way we work?
+-   Frequency: quarterly
+-   Responsible: Scrum master
+-   Contributors: entire engineering team
+-   Products:
+    -   Ranked list of things that worked well and things that didn't
+    -   Ranked list of ideas for things to try to adopt, improve, or drop
+
+> Note: our product manager has been doubling as our scrum master for the last couple of years.
+> Those jobs are usually done by different people in larger organizations,
+> so the description above lists the role rather than the person who happens to be doing it right now.
+
+### Engineering All Hands
+
+Our company includes biologists working in a wet lab,
+data scientists,
+and machine learning specialists staring into the void and hoping it blinks first.
+And while most of us are in Toronto,
+a substantial minority live and/or work remotely.
+This meeting is where the VP of Engineering updates us all
+on things we might otherwise miss
+(both organizationally and technically).
+
+-   Summary: Biweekly team sync to discuss company and/or engineering news
+-   Key Questions:
+    -   What's going on elsewhere in the company that might affect us?
+    -   What's coming up that we ought to know about?
+-   Frequency: biweekly
+-   Responsible: VP Engineering
+-   Contributors: entire engineering team plus invited guests
+-   Products:
+    -   Heightened awareness of recent and upcoming changes
+
+### Engineering Design Reviews
+
+One of the things that has come out of our end-of-quarter reviews
+is a greater emphasis on presenting designs for new features or applications to the whole team
+before diving into implementation.
+The goal here is *not* to create a pile of UML diagrams that no one will ever look at again,
+but rather to explain what the problem actually is,
+what factors a solution must take into account,
+what alternatives there are,
+and what we're *not* going to do.
+Some designs go through two or three overhauls before work starts;
+in almost every case,
+checking our thinking like this has saved us pain downstream.
```

Comment:
Might be useful to give an example of what pain this has saved us e.g. having to rebuild or rearchitect a solution half-way through delivery